src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1111,14 +1111,17 @@ private void InitializePackagesFolderInfo(string packagesFolderPath, bool exclud
 
                 var installedPackagesInDependencyOrder = await GetInstalledPackagesInDependencyOrder(nuGetProject, token);
 
+                var isDependencyBehaviorIgnore = resolutionContext.DependencyBehavior == DependencyBehavior.Ignore;
```

Comment:
Why map the DependencyBehavior enum to a bool? Couldn't you pass it all the way down and make this `==` check at the last moment?